src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -351,13 +374,63 @@ public AzureSynapsePdao(
     this.synapseJdbcTemplate = synapseJdbcTemplate;
   }
 
+  /**
+   * Initialize a Synapse database with the given name and encryption key. Note: we need to connect
+   * to the default `master` database to create the new database.
+   */
+  @PostConstruct
+  private void initializeDb() {
+    // If this configuration wasn't set, skip initialization
+    if (azureResourceConfiguration.getSynapse() == null) {
+      return;
+    }
+    boolean initialize = azureResourceConfiguration.getSynapse().isInitialize();
+    String dbName = azureResourceConfiguration.getSynapse().getDatabaseName();
+    String encryptionKey = azureResourceConfiguration.getSynapse().getEncryptionKey();
+    String parquetFormatName = azureResourceConfiguration.getSynapse().getParquetFileFormatName();
+
+    if (initialize) {
+      logger.info("Initializing Synapse database {}", dbName);
+      SQLServerDataSource dsInit = getDatasource(DEFAULT_DB_NAME);
+      try (Connection connection = dsInit.getConnection();
+          Statement statement = connection.createStatement()) {
+        statement.execute(new ST(DB_CREATION_TEMPLATE).add("dbname", dbName).render());
+      } catch (SQLException e) {
+        throw new PdaoException("Error creating database", e);
+      }
+
+      // Connect to the newly created db to set up encryption
+      SQLServerDataSource ds = getDatasource();
```

Comment:
It seems odd to me to use `dbName` above and then call `getDatasource()` here as it re-derives the database name from the config. Why not call `getDatasource(dbName)`? I think it would make this clearer.

Reply:
Oh, that's because we need to connect with a different database (because the one we are creating doesn't exist yet)

Reply:
Not sure what you mean. Calling `getDatasource(dbName)` has the same effect as calling `getDatasource()`. That's because `getDatasource()` is implemented as:
```
  public SQLServerDataSource getDatasource() {
    return getDatasource(azureResourceConfiguration.getSynapse().getDatabaseName());
  }
```
And `dbName` is defined using `azureResourceConfiguration.getSynapse().getDatabaseName()`. So the two forms are equivalent. But using the form `getDatasource(dbName)` makes the intent of this code clearer; it's first creating the database and then it's configuring it. Calling `getDatasource()` obscures the relationship between the code that creates the database and this code.